src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/OOPTagHelperResolver.cs

```diff
@@ -146,13 +161,24 @@ public override async Task<TagHelperResolutionResult> GetTagHelpersAsync(Project
             {
                 // Not a delta based response, we should treat it as a "refresh"
                 tagHelpers = Array.Empty<TagHelperDescriptor>();
+                fromCache = false;
             }
 
             if (deltaResult.ResultId != lastResultId)
             {
                 // New results, lets build a coherent TagHelper collection and then cache it
                 tagHelpers = deltaResult.Apply(tagHelpers);
                 _resultCache.Set(projectFilePath, deltaResult.ResultId, tagHelpers);
+                fromCache = false;
+            }
+
+            stopWatch.Stop();
+            if (fromCache)
+            {
+                _telemetryReporter.ReportEvent("taghelpers.fromcache", VisualStudio.Telemetry.TelemetrySeverity.Normal, new Dictionary<string, long>()
+                {
+                    { "taghelper.cachedresult.ellapsedms", stopWatch.ElapsedMilliseconds }
```

Comment:
ditto about number found? the cache might be horrible ineffective, and not worth the complexity, for all we know